tools/operator/connectionprofile/connectionprofile.go

```diff
@@ -91,6 +100,221 @@ func (c ConnProfile) GenerateConnProfilePerOrg(orgName string) error {
+	}
+	userCertsPath := paths.JoinPath(signCertsPath, fmt.Sprintf("%s", signCertsFile[0].Name()))
+	cert, err = c.GetCertificateFromFile(userCertsPath)
+	if err != nil {
+		return clients, err
+	}
+	client.Client.ClientSignedCert.Pem = cert
+	clientName := fmt.Sprintf("client-%s", orgName)
+	clients[clientName] = client
+
+	return clients, nil
+}
+
+func (c ConnProfile) channelCaliper(orgName string, cprofile CaliperConnProfile) (map[string]networkspec.CaliperChannel, error) {
+	channels := make(map[string]networkspec.CaliperChannel)
+
+	// for each channel
+	for i := 0; i < c.Config.NumChannels; i++ {
+		var channel networkspec.CaliperChannel
+		//channel created
+		channel.Created = true
+
+		//orderers
+		ordererMap := cprofile.Orderers
+		for o := range ordererMap {
+			channel.Orderers = append(channel.Orderers, o)
+		}
+
+		//peers
+		peerMap := cprofile.Peers
+		pr := make(map[string]networkspec.CaliperChannelPeer)
+		for pName := range peerMap {
+			var p networkspec.CaliperChannelPeer
+			p.EventSource = true
+			pr[pName] = p
+		}
+		channel.Peers = pr
+
+		//chaincode
+		// for chaincodes loop
+		for j := 0; j < len(c.Config.ChaincodeIDs); j++ {
+			var cc networkspec.ChaincodeID
+			cc.Id = c.Config.ChaincodeIDs[j].Id
+			cc.Version = c.Config.ChaincodeIDs[j].Version
+			channel.Chaincodes = append(channel.Chaincodes, cc)
+		} // end chaincodeIDs loop
+
+		chName := c.Config.ChannelPrefix
+		if len(chName) == 0 {
+			chName = "testorgschannel"
+		}
+		chName = fmt.Sprintf("%s%d", chName, i)
+		channels[chName] = channel
+	} // end channel loop
+
+	return channels, nil
+}
+
+func (c ConnProfile) GenerateCaliperConnProfilePerOrg(orgName string) error {
+
+	var err error
+	path := paths.CaliperConnectionProfilesDir(c.Config.ArtifactsLocation)
+	fileName := paths.JoinPath(path, fmt.Sprintf("caliper_connection_profile_%s.yaml", orgName))
+
+	cp := ConnProfile{Organizations: c.Organizations, Orderers: c.Orderers, Peers: c.Peers, CA: c.CA}
+	cprofile, err := c.connprofile2caliper(cp)
+	if err != nil {
+		logger.ERROR("Failed to convert connection profile to Caliper")
+		return err
+	}
+
+	caliper := networkspec.Caliper{}
+	caliper.Blockchain = "fabric"
+
+	clients, err := c.clientCaliper(orgName)
+	if err != nil {
+		logger.ERROR("Failed to get the clients for Caliper connection profile")
+		return err
+	}
+
+	channels, err := c.channelCaliper(orgName, cprofile)
+	if err != nil {
+		logger.ERROR("Failed to get the channels for Caliper connection profile")
+		return err
+	}
+
+	ccp := networkspec.CaliperConnectionProfile{Caliper: caliper, Clients: clients, Channels: channels, Organizations: cprofile.Organizations, Orderers: cprofile.Orderers, Peers: cprofile.Peers, CA: cprofile.CA}
+	yamlBytes, err := yaml.Marshal(ccp)
+	if err != nil {
+		logger.ERROR("Failed to convert the connection profile struct to bytes")
+		return err
+	}
+	_, err = os.Create(fileName)
+	if err != nil {
+		logger.ERROR("Failed to create ", fileName)
+		return err
+	}
```

Comment:
no need to initialize the file, `ioutil.Writefile` will create it if it does not exist

Reply:
ok